v2/devdoc/logging.md

```diff
@@ -0,0 +1,162 @@
+# `c-logging` requirements
+
+## Basic functionality
+
+`c-logging` shall expose APIs allowing the user to log events that can be used for observing the software activity, debugging, etc.
+
+The exposeed API shall allow creating contexts both dynamically and on the stack and using those contexts for logging.
+
+The API shall allow easy formatting of values to be placed in the contexts by using printf-like formatting.
+
+```
+void logger_log(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* format, ...);
+
+// macro that can be used to create a dynamically allocated context
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    ...
+
+// destroy a dynamically allocated context
+#define LOG_CONTEXT_DESTROY(log_context_handle) \
+    ...
+
+// macro that can be used to create a context on the stack
+#define LOCAL_LOG_CONTEXT_DEFINE(log_context, parent_context, ...) \
```

Comment:
>LOCAL_LOG_CONTEXT_DEFINE [](http://example.com/codeflow?start=8&length=24)

poetry: let's name this LOG_CONTEXT_LOCAL_DEFINE.

Reply:
Nothing that a rename cannot solve, done!